bedrock/src/primitives/mod.rs

```diff
@@ -0,0 +1,87 @@
+/// A wrapper around a hex-encoded string.
+///
+/// This is used to ensure that the hex string is properly formatted and it's always prefixed with "0x".
+/// Ensures consistency in expected output.
+///
+/// # Examples
+/// ```
+/// use bedrock::primitives::HexEncodedString;
+/// let hex_string = HexEncodedString::new("0x1234567890abcdef".to_string());
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, uniffi::Object)]
+pub struct HexEncodedString(String);
+
+#[uniffi::export]
+impl HexEncodedString {
+    /// Initializes a new `HexEncodedString` from a hex string.
+    ///
+    /// # Arguments
+    /// * `s` - The hex string to initialize the `HexEncodedString` from. May or may not be prefixed with "0x".
+    ///
+    /// # Errors
+    /// - `PrimitiveError::InvalidHexString` if the provided string is not validly encoded hex data.
+    #[uniffi::constructor]
+    pub fn new(s: String) -> Result<Self, PrimitiveError> {
+        let mut s = s;
+
+        if s.starts_with("0x") {
+            s = s[2..].to_string();
+        }
+
+        hex::decode(&s).map_err(|_| PrimitiveError::InvalidHexString(s.clone()))?;
+
+        Ok(Self(format!("0x{s}")))
+    }
+
+    /// Returns the wrapped hex string as a string.
+    #[must_use]
+    pub fn as_string(&self) -> String {
+        self.0.clone()
+    }
+}
+
+impl From<HexEncodedString> for String {
+    fn from(hex_encoded_string: HexEncodedString) -> Self {
+        hex_encoded_string.0
+    }
+}
+
```

Comment:
we can also implement `Display`. In the future, I can imagine we can also implement `From` converters from alloy types